account-kit/logging/src/index.ts

```diff
@@ -1,15 +1,54 @@
 import { createClientLogger } from "./client.js";
 import { createServerLogger } from "./server.js";
-import type { EventLogger, EventsSchema } from "./types";
+import type { EventLogger, EventsSchema, LoggerContext } from "./types";
 
 export type * from "./types.js";
 
-export function createLogger<
-  Schema extends EventsSchema
->(): EventLogger<Schema> {
-  if (typeof window === "undefined") {
-    return createServerLogger();
-  }
+export function createLogger<Schema extends EventsSchema = []>(
+  context: LoggerContext
+): EventLogger<Schema>;
 
-  return createClientLogger();
+export function createLogger(context: LoggerContext): EventLogger {
+  const innerLogger =
+    typeof window === "undefined"
+      ? createServerLogger(context)
+      : createClientLogger(context);
+
+  const logger: EventLogger = {
+    ...innerLogger,
+    profiled<TFunc extends (...args: any[]) => Promise<unknown> | unknown>(
```

Comment:
I wanna log performance metrics for certain methods and this makes it super easy to wrap existing functions with profiling